apps/prairielearn/src/lib/db-types.ts

```diff
@@ -553,7 +553,7 @@ export const SubmissionSchema = z.object({
   mode: z.enum(['Public', 'Exam', 'SEB']).nullable(),
   override_score: z.number().nullable(),
   params: z.record(z.string(), z.any()).nullable(),
-  partial_scores: z.record(z.string(), z.number()).nullable(),
+  partial_scores: z.record(z.string(), z.any()).nullable(),
```

Comment:
Partial scores actually typically points to a record of string to an object that has score/weight/feedback, though I don't think this should be enforced.